compiler/resolution/implementForallIntents.cpp

```diff
@@ -694,6 +695,23 @@ static void verifyOuterVars(BlockStmt* body2,
 //
 void implementForallIntents1(DefExpr* defChplIter)
 {
+  // Sometimes defChplIter is introduced for ForallStmt. If so, ignore.
+  bool oldStyle = false;
+  if (DefExpr* next = toDefExpr(defChplIter->next))
+    if (!strcmp(next->sym->name, "chpl__leadIter") ||
+        !strcmp(next->sym->name, "chpl__saIter"))
```

Comment:
It's a shame to use strcmps for this. Seems like a flag would be better. However I'm not objecting too strongly if you expect to remove "old style" foralls in short order.